### PR TITLE
Update CLR ABI PSPSym Section

### PR DESCRIPTION
**Pull request by parjong**
This commit updates CLR ABI PSPSym Section according to recent changes on x86/Linux.

### CODE REVIEW

File: Documentation/botr/clr-abi.md [hunk 1]
```diff
@@ -273,15 +273,19 @@ Note that JIT64 does not implement this properly. The C# compiler used to always
...
+The *PSPSym* (which stands for Previous Stack Pointer Symbol) is a pointer-sized local variable used to access locals from the main function body.
+
+CoreRT does not use PSPSym. For filter funclets the VM sets the frame register to be the same as the parent function. For second pass funclets the VM restores all non-volatile registers. The same convention is used across all platforms.
+
+CoreCLR uses PSPSym for all platforms except x86: the frame pointer on x86 is always preserved when the handlers are invoked.
 
 First, two definitions.
 
 *Caller-SP* is the value of the stack pointer in a function's caller before the call instruction is executed. That is, when function A calls function B, Caller-SP for B is the value of the stack pointer immediately before the call instruction in A (calling B) was executed. Note that this definition holds for both AMD64, which pushes the return value when a call instruction is executed, and for ARM, which doesn't. For AMD64, Caller-SP is the address above the call return address.
```
Comment (parjong) [inline Documentation/botr/clr-abi.md]: > For AMD64, Caller-SP is the address above the call return address.

Is it correct? I found this line while working on this commit.

**Comment by parjong** [comment]
@dotnet-bot test OSX x64 Checked Build and Test please

File: Documentation/botr/clr-abi.md (hunk 1 above)
Comment (janvorli) [inline Documentation/botr/clr-abi.md]: @parjong why doesn't it sound right to you?

Comment (parjong) [inline Documentation/botr/clr-abi.md]: For me, this sentence seems to say that ``Caller-SP`` on AMD64 is ``CFA - 8``. Is that true?

Comment (janvorli) [inline Documentation/botr/clr-abi.md]: No, CFA is the SP before the call is made, so caller SP is CFA.

Comment (parjong) [inline Documentation/botr/clr-abi.md]: Does ``the address above the call return address`` mean that ``Caller-SP`` is equal to the address where ``rip`` saved + 8? If so, it seems correct.

Comment (janvorli) [inline Documentation/botr/clr-abi.md]: Yes, it means that. What was your former understanding?

Comment (parjong) [inline Documentation/botr/clr-abi.md]: I thought that it means ``CFA(call return address) -(above) 8``

**Comment by parjong** [comment]
@dotnet-bot test OSX x64 Checked Build and Test please

**Comment by BruceForstall** [comment]
LGTM.

BTW, please tag me for any CLR ABI doc changes.

**Comment by BruceForstall** [comment]
@parjong It seems like the main change here is:
```
CoreCLR uses PSPSym for all platforms except x86: the frame pointer on x86 is always preserved when the handlers are invoked.
```
So the first part is true:
```
CoreCLR uses PSPSym for all platforms except x86
```
but the 2nd part I'm not sure of, or is confusing:
```
the frame pointer on x86 is always preserved when the handlers are invoked.
```
Is this true for both Windows/x86 and Linux/x86? Should we add some text to clarify the behavior of Linux/x86 compared to Windows/x86?

**Comment by parjong** [comment]
@BruceForstall I do not know how x86/Windows works. Is it similar to Core RT? The approach that x86/Linux currently seems to be similar to that of Core RT (VM restores frame pointer for filter, and all the non-volatile resisters for the second pass funclets).

**Comment by BruceForstall** [comment]
@parjong I'm not positive myself. Since this is in the PSPSym section, which does not apply to Windows/x86, I'll go ahead and merge the change.

**Pull request merged**